great_expectations/core/batch.py

```diff
@@ -706,7 +715,7 @@ def get_batch_request_as_dict(
     if batch_request is None:
         return None
 
-    if isinstance(batch_request, (BatchRequest, RuntimeBatchRequest)):
+    if isinstance(batch_request, BatchRequestBase):
```

Comment:
This is the only runtime change in the PR.
It seems obvious at first but the docstring for `BatchRequestBase` makes me question this.

>    This class is for internal inter-object protocol purposes only.
    As such, it contains all attributes of a batch_request, but does not validate them.
    See the BatchRequest class, which extends BatchRequestBase and validates the attributes.

> BatchRequestBase is used for the internal protocol purposes exclusively, not part of API for the developer users.